torch/csrc/jit/codegen/cuda/lower_index.cpp

```diff
@@ -1157,8 +1162,8 @@ void IndexLowering::handleGroupedGridWelford(
 }
 
 void IndexLowering::handle(const LoadStoreOp* ldst) {
-  const auto in = lowerSrcIndex(ldst->in(), ldst->out());
-  const auto out = lowerDstIndex(ldst->out());
+  const auto in = lowerSrcIndex(ldst->in(), ldst->out(), {}, true);
```

Comment:
> I also wonder what happens if we enable this for all smem tensors. Wouldn't it be better?

Is it possible to dereference the `unsigned` shared memory address and get the value there?

Reply:
> Is `cvta_smem_address` true only in this case?

Yes

Reply:
> I wonder if the decision could be done inside the index compute routines as they know the defining and using expressions of given producer and consumer tensors.

No, you can not. The definition of the consumer being a `LoadStoreOp` does not necessarily means the current expression we are lowering should do `cvta`. For example, in the initialization of the shared memory buffer, the definition of consumer is a `LoadStoreOp`, but the expr we are handling is a `arraySet`, which should user pointer instead of `unsigned` for indexing.

Reply:
> > I also wonder what happens if we enable this for all smem tensors. Wouldn't it be better?
> 
> Is it possible to dereference the `unsigned` shared memory address and get the value there?

Ah, OK, we could `reinterpret_cast`, but don't know what assembly code would be generated. I thought it might help as SMEM addresses are guaranteed to be within the 32-bit range.

Reply:
> > I wonder if the decision could be done inside the index compute routines as they know the defining and using expressions of given producer and consumer tensors.
> 
> No, you can not. The definition of the consumer being a `LoadStoreOp` does not necessarily means the current expression we are lowering should do `cvta`. For example, in the initialization of the shared memory buffer, the definition of consumer is a `LoadStoreOp`, but the expr we are handling is a `arraySet`, which should use pointer instead of `unsigned` for indexing.

OK, makes sense.